traits/ctraits.c

```diff
@@ -4269,7 +4273,7 @@ _trait_set_validate(trait_object *trait, PyObject *args)
                     break;
 
                 case 22: /* Callable check: */
-                    if (n == 1) {
+                    if (n == 2) {
```

Comment:
I'm wondering whether there's a potential pickle compatibility problem here: if a `Callable` trait has been pickled with Traits 6.0.0, does this check prevent it from being unpickled?

Reply:
That will be a problem since the expected number of arguments is different, I currently don't ensure that there are 2 arguments before I index into the second one.  I think i'll count the args and also change the above to `n <=2 ` and then choose the correct flow accordingly.